src/sidebar/services/groups.js

```diff
@@ -213,9 +213,11 @@ function groups(
         }
         return Promise.all(groupApiRequests).then(
           ([myGroups, featuredGroups, token, selectedGroup]) => [
+            // Don't add the selectedGroup if it's already in the featuredGroups.
             combineGroups(
               myGroups,
-              selectedGroup !== undefined
+              selectedGroup !== undefined &&
```

Comment:
I wasn't sure if it was better to put this logic here or push it into the combineGroups function. The pro's of having this here is the combineGroups is responsible for determining scope and user membership of the group and combining the two groups lists: featuredGroups and myGroups (technically the directLinkedGroup/directLinkedAnnGroup can go into one of the already existing groups lists which simplifies the inputs to the combineGroups function). The con's this adds more logic to the load in the groups service because it has to only put the group in there under certain circumstances as opposed to passing the directLinkedGroup and directLinkedAnnGroup and letting the combineGroups sort out when to add them (if any). The third option would be to de-dup the featuredGroups list in combineGroups but I feel like that's making an assumption about the input that doesn't necessarily make sense (that the groups in the passed list could be duplicated).

Reply:
> The third option would be to de-dup the featuredGroups list

Third option? I'm not entirely clear on how this differs from the first two. By "push it into the combineGroups function" did you mean adding an extra argument to `combineGroups`?

Honestly I don't think it makes that much difference at present. If there were several more sources of groups being combined then I think it would become simpler to de-duplicate in `combineGroups`,  because you could just add a few lines at the end to de-duplicate the final `groups` array.

Reply:
In the third option I mean add the group into the featuredGroups list so there are two of the same group and just add a de-dup in the combineGroups function.